service-commands/scripts/setup-instance.sh

```diff
@@ -17,14 +17,15 @@ PROTOCOL_DIR=${PROTOCOL_DIR:-$(dirname $(realpath $0))/../../}
 source $PROTOCOL_DIR/service-commands/scripts/utils.sh
 
 # parse arguments
-while getopts "p:u:c:r:l:f" flag; do
+while getopts "p:u:c:r:l:f:s:" flag; do
```

Comment:
should be `"p:u:c:r:l:fs:"`

Reply:
why is there no colon between f and s?

Reply:
`:` means that the previous option, in this case `f` has an argument, here `f` does not have an argument and is a boolean. Therefore we should not have a `:` here.

Reply:
Looks like `s` is also a boolean, and you can remove the `:` after that too. This would mean that other parts of the script need to be updated to account for this.

However, I think we should fix the first one, that does not require any further code changes. Since the otherwise the script might break when provisioning under fast.

Reply:
i also tested the cases with/without spot and fast and they work together fine